db/migrate/20191023180330_migrate_ems_ref_obj_to_ems_ref_type.rb

```diff
@@ -0,0 +1,61 @@
+class MigrateEmsRefObjToEmsRefType < ActiveRecord::Migration[5.1]
+  class EmsCluster < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled # disable STI
+    include ActiveRecord::IdRegions
+  end
+  class EmsFolder < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled # disable STI
+    include ActiveRecord::IdRegions
+  end
+  class Host < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled # disable STI
+    include ActiveRecord::IdRegions
+  end
+  class ResourcePool < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled # disable STI
+    include ActiveRecord::IdRegions
+  end
+  class Snapshot < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled # disable STI
+    include ActiveRecord::IdRegions
+  end
+  class Storage < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled # disable STI
+    include ActiveRecord::IdRegions
+  end
+  class VmOrTemplate < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled # disable STI
+    self.table_name         = "vms"
+    include ActiveRecord::IdRegions
+  end
+
+  MODELS_WITH_EMS_REF_OBJ = [EmsCluster, EmsFolder, Host, ResourcePool, Snapshot, Storage, VmOrTemplate].freeze
+
+  def up
+    MODELS_WITH_EMS_REF_OBJ.each do |klass|
+      say_with_time("Converting ems_ref_obj to ems_ref_type for #{klass.name}") do
+        klass.in_my_region.where("ems_ref_obj LIKE '%VimString%'").find_each do |obj|
+          ems_ref_obj  = YAML.load(obj.ems_ref_obj)
+          ems_ref_type = ems_ref_obj.vimType
+          obj.update!(:ems_ref_type => ems_ref_type)
+        end
+      end
+    end
+  end
+
+  def down
+    MODELS_WITH_EMS_REF_OBJ.each do |klass|
+      say_with_time("Converting ems_ref_type to ems_ref_obj for #{klass.name}") do
+        klass.in_my_region.find_each do |obj|
+          ems_ref_obj = if obj.ems_ref_type.nil?
+            obj.ems_ref
+          else
+            VimString.new(obj.ems_ref, obj.ems_ref_type, :ManagedObjectReference)
```

Comment:
Same here 😆 .

We can always manually construct the YAML if needed.

Alternately, we could copy the classes into this migration though that may actually make things worse.

Reply:
Same as above, that is the plan